### ISSUE TITLE
Capslock Notice

### ISSUE THREAD

**Issue by xenonca** [minetest/minetest#7843]
##### Issue type

 - Feature request
 
##### Minetest version

```
4.17.1
```

##### OS / Hardware

Operating system: Ubuntu 18.04

##### Summary

I think it would be quite useful, if users would get a little notice in the password section (maybe a small exclamation mark) when capslock is activated.

**Comment by paramat** [comment]
However consider that passwords display as '*********' so that there is no visible information showing what the password is. A capslock notice is visual information that narrows down what the password is. So technically is a little worse for security.

**Comment by xenonca** [comment]
Though that notice would only have to show when caps lock is active, independently from the password itself having caps in it or not.

**Comment by paramat** [comment]
Yeah i admit it's a very minor security risk, but is still wrong in principle.

**Comment by ClobberXD** [comment]
There's also the issue of the whole formspec having to be updated for even minor modifications. It'd be a pain to have the password field reset mid-way, just to have a caps-lock notice displayed.

The feature would be nice to have, though. It's just the implementation that needs some thinking over.

**Comment by rubenwardy** [comment]
This would be properly implemented as an extension to irrtextelement

**Comment by xenonca** [comment]
Ok, just thought it'd be a nice feature, wasn't aware of the effort it would take (and the security issue)

**Comment by Ryan-Nolan** [comment]
Could this not be implemented as a Client Side Mod? so essentially it becomes optional and down to user preference?  What engine changes would be needed to make a CSM version of this a possibility?

**Comment by Ryan-Nolan** [comment]
such a shame, that your opinion is more important than everyone elses.....  maybe one day you might learn to talk about how things can be done rather than what cant be done. "especially when it concens ideas that you didnt come up with". im getting bored of reading the same comments from paramat day in day out.

**Comment by rubenwardy** [comment]
This cannot be done as a Client-Side Mod. The correct implementation is using a derived Irrlicht GUI element

**Comment by Ryan-Nolan** [comment]
thanks for the answer @rubenwardy that's fine... it was just a suggestion.. 😄 lets say the CSM wasnt as doomed as paramat makes out.... would it be impossible for the CSM to be modified engine side to accomodate such a mod?

**Comment by rubenwardy** [comment]
Yes, it would be impossible. Client-side mods do not run in the main menu

**Comment by paramat** [comment]
Ryan-Nolan, my opinion is not more important than anyone elses, i haven't stated that anywhere, i'm just stating my opinions as any core dev does and it's obvious that they are simply my opinions.
I'm sorry that we disagree on many issues, when this happens with someone i actually feel bad about it, because it is not something personal.

Unprovoked, you are being insulting, and made baseless accusations about what i wrote. At any point i'm ready to go forward and get on with you, so it's up to you. If my opinions bother you so much you don't have to read them.

**Comment by Ryan-Nolan** [comment]
@paramat Not sure why I got a notification to this, I thought i unsubscribed from all things Minetest related on GitHub because admittedly your behavior on here was annoying me more than it should... To clarify, my issue is not with your opinion but with the way you project it. What i mean by that is you don't let anyone else express their opinions unless it conforms with your own. you attempt to shut them down almost every time... simply put, you don't need to repeat your opinion over and over again. Like is said above, I'm going to avoid Minetest on GitHub as much as i can, hopefully that will avoid these situations.

**Comment by SmallJoker** [comment]
TL;DR. I would welcome to see a visual hint in password boxes that caps lock is turned on. Displaying an icon as suggested in the main post would be the best solution for this request. Using CSM here is pointless since that would not work for server sent formspecs (with password fields).

**Comment by Zughy** [comment]
Can't we just check in `pwdfield`s if the caps lock is on and if it is, display a lock icon inside the field (on the right)?

**Comment by rubenwardy** [comment]
This doesn't decrement security by much. It only protects against observers IRL, and doesn't give them enough meaningful information to discover the password. They would still need to see the keys. I imagine most users would use the same password on every server, anyway

So, I don't object to this and I think it's a good idea. But I also think that there are other things that should take higher priority when it comes to UI/UX
